### DISCUSSION TITLE
tsconfig.json configuration

### DISCUSSION THREAD

**Discussion by jrson83** [nanojsx/nano#64] [Help / Questions] [answered]
Hey,
I'm trying to understand the different configs of tsconfig.json in the [template](https://github.com/nanojsx/template) and [example1](https://github.com/nanojsx/example1).

Could you explain `jsxImportSource: nano-jsx/lib` and `jsxFactory: Nano.h` and provide some information on why you use different configuration of `target`, `module` and `jsx`?

**Comment by yandeu** [comment, marked as answer]
Please read:
 - https://www.typescriptlang.org/docs/handbook/tsconfig-json.html
 - https://www.typescriptlang.org/tsconfig

**Reply by jrson83** [reply]
Thanks, now I also understand why there was sometimes errors related to VS Code and HTML-Elements.

**Comment by btiernay** [comment]
Similar to the OP's question, I'm noticing several ways to configure typescript when using this project:
- https://github.com/nanojsx/nanojsx.github.io/blob/e58ff9bd7a378008c81c0ab471fb5b918953d860/js/code-docs.js#L19-L21
- https://github.com/nanojsx/node-jsx-example/blob/main/tsconfig.json#L10-L11
- https://github.com/nanojsx/nano/blob/master/tsconfig.json#L9-L10
- https://github.com/nanojsx/nano/blob/master/tsconfig.test.jsx-runtime.json#L9-L10
- https://github.com/nanojsx/template/blob/master/tsconfig.json#L9-L11
- https://github.com/nanojsx/example1/blob/main/tsconfig.json#L9-L10

Which is correct with the latest versions? Some of these don't make much sense, so I am a bit confused. Thanks!

**Reply by yandeu** [reply]
In version 0.0.36 you can use:

for commonjs:
```json
{
    "jsx": "react-jsx",
    "jsxImportSource": "nano-jsx/lib"
}
```

for esm:
```json
{
    "jsx": "react-jsx",
    "jsxImportSource": "nano-jsx/esm"
}
```
both replace the `h`

**Reply by MarvinMiles** [reply]
@yandeu hi!
Cant make it work with https://nitro.unjs.io/.  Getting `React is not defined` error. Tried with jsxFactory option as well.

Compiled output is
```renderSSR(() => /* @__PURE__ */ React.createElement("div", null, "Hello"));```

Maybe you can help me? Or should I make an issue in https://github.com/unjs/nitro?

**Reply by yandeu** [reply]
Hmmm. Yes, maybe it's best to open an issue on nitro.

**Reply by MarvinMiles** [reply]
In case if someone will be interested to use nano-jsx with nitro, I made a [PR](https://github.com/unjs/nitro/pull/1303) which was just merged.
Now everything works. Moreover, from now nano-jsx example presents officially in nitro repository https://github.com/unjs/nitro/tree/main/examples/nano-jsx

**Reply by yandeu** [reply]
Cool thanks!

**Comment by btiernay** [comment]
Thanks! Might want to consider updating the docs to match:
<img width="829" alt="image" src="https://user-images.githubusercontent.com/2717578/211230809-71f6e62c-1d35-48f1-a2f5-069430443cfb.png">
